vanilla/src/main/java/org/spongepowered/vanilla/server/packs/PluginPackResources.java

```diff
@@ -89,12 +89,15 @@ public Collection<ResourceLocation> getResources(final PackType type, final Stri
             final Predicate<String> fileNameValidator) {
         try {
             final Path root = this.typeRoot(type);
-            return Files.walk(root.resolve(namespace).resolve(namespace), depth)
-                .filter(s -> !s.getFileName().toString().endsWith(".mcmeta"))
-                .map(Object::toString)
-                .filter(fileNameValidator)
-                .map(s -> new ResourceLocation(namespace, path))
-                .collect(Collectors.toList());
+            final Path namespaceDir = root.resolve(namespace);
+            return Files.walk(namespaceDir, depth)
+                    .filter(Files::isRegularFile)
```

Comment:
The `FilePackResources` from Minecraft looks like below codes, so I filter files:
```
while(var3.hasMoreElements()) {
    ZipEntry var7 = var3.nextElement();
    if (!var7.isDirectory()) {
        String var8 = var7.getName();
        if (!var8.endsWith(".mcmeta") && var8.startsWith(var6)) {
            String var9 = var8.substring(var5.length());
            String[] var10 = var9.split("/");
            if (var10.length >= param3 + 1 && param4.test(var10[var10.length - 1])) {
                var4.add(new ResourceLocation(param1, var9));
            }
        }
    }
}
```